cmd/infra/aws/create.go

```diff
@@ -1,225 +1,395 @@
+	keys = append(keys, jose.JSONWebKey{
+		Key:       pubKey,
+		KeyID:     kid,
+		Algorithm: string(jose.RS256),
+		Use:       "sig",
+	})
+
+	type KeyResponse struct {
+		Keys []jose.JSONWebKey `json:"keys"`
 	}
-	r53client, err := route53Client(o.AWSCredentialsFile)
+	jwks, err := json.MarshalIndent(KeyResponse{Keys: keys}, "", "  ")
 	if err != nil {
-		return nil, err
+		return fmt.Errorf("failed to marshal KeyResponse: %w", err)
 	}
-	result.PublicZoneID, err = o.LookupPublicZone(r53client)
+
+	_, err = s3Client.PutObject(&s3.PutObjectInput{
+		ACL:    aws.String("public-read"),
+		Body:   bytes.NewReader(jwks),
+		Bucket: aws.String(bucketName),
+		Key:    aws.String("openid/v1/jwks"),
+	})
 	if err != nil {
-		return nil, err
+		return fmt.Errorf("failed to put jwks in bucket: %w", err)
 	}
-	result.PrivateZoneID, err = o.CreatePrivateZone(r53client, result.VPCID)
+	log.Info("JWKS document updated", "bucket", bucketName)
+
+	discoveryTemplate := `{
+	"issuer": "%s",
+	"jwks_uri": "%s/%s",
+	"response_types_supported": [
+		"id_token"
+	],
+	"subject_types_supported": [
+		"public"
+	],
+	"id_token_signing_alg_values_supported": [
+		"RS256"
+	],
+	"claims_supported": [
+		"aud",
+		"exp",
+		"sub",
+		"iat",
+		"iss",
+		"sub"
+	]
+}`
+
+	discoveryJSON := fmt.Sprintf(discoveryTemplate, issuerURL, issuerURL, "openid/v1/jwks")
+	_, err = s3Client.PutObject(&s3.PutObjectInput{
+		ACL:    aws.String("public-read"),
+		Body:   aws.ReadSeekCloser(strings.NewReader(discoveryJSON)),
+		Bucket: aws.String(bucketName),
+		Key:    aws.String(".well-known/openid-configuration"),
+	})
 	if err != nil {
-		return nil, err
+		return fmt.Errorf("failed to put discovery JSON in bucket: %w", err)
 	}
-	return result, nil
+	log.Info("OIDC discovery document updated", "bucket", bucketName)
+
+	return nil
 }
 
-func ec2Client(creds, region string) (ec2iface.EC2API, error) {
-	awsConfig := &aws.Config{
-		Region: aws.String(region),
-	}
-	awsConfig.Credentials = credentials.NewSharedCredentials(creds, "default")
-	s, err := session.NewSession(awsConfig)
+// TODO: Replace direct Route53 manipulation with CloudFormations
+func (o *CreateInfraOptions) configureSubdomain(ctx context.Context, cf *cloudformation.CloudFormation, r53 route53iface.Route53API, stackID string) error {
+	var nameservers, zoneID, subdomain string
+	err := wait.PollUntil(5*time.Second, func() (bool, error) {
+		stack, err := getStack(cf, stackID)
+		if err != nil {
+			log.Error(err, "failed to get stack", "id", stackID)
+			return false, nil
+		}
+		nameservers = getStackOutput(stack, "SubdomainPublicZoneNameServers")
+		subdomain = getStackOutput(stack, "Subdomain")
+		zoneID = getStackOutput(stack, "BaseDomainHostedZoneId")
+		if len(nameservers) > 0 {
+			return true, nil
+		}
+		return false, nil
+	}, ctx.Done())
 	if err != nil {
-		return nil, fmt.Errorf("failed to create client session: %w", err)
+		return fmt.Errorf("failed waiting for subdomain nameserver output")
```

Comment:
It'd be nice to store the real error preventing this from proceeding, e.g last error returned by `stack, err := getStack(cf, stackID)` or `nameservers is empty` and return it here